src/Modes.jl

```diff
@@ -322,11 +322,13 @@ coordinates `xs = (r,θ)`.
 """
 function overlap(m::AbstractMode, E)
     dl = dimlimits(m)
-    function f(xs)
-        0.5*sqrt(ε_0/μ_0)*dot(conj(Exy(m, xs)), E(xs))*xs[1]
+    rval, _ = hcubature(dl[2], dl[3]; maxevals=1000) do xs
```

Comment:
Does `hcubature` not handle the complex case? The C library does.

Reply:
It doesn't see https://github.com/JuliaMath/Cubature.jl/issues/21 As far as I can tell the C library only supports it through using the vector interface, but maybe I'm wrong? Anyway here we can use the pure Julia implementation instead and it works fine without splitting.

Reply:
Yes, the vector interface but with pairwise error. Anyway, the Julia solution is better.